### ISSUE TITLE
Cannot register new user

### ISSUE THREAD

**Issue by blankbeard** [fethica/PHP-Login#50]
I am getting the error 'An error occurred... try again' on 'login/signup.php'. From my understanding, the error is coming from 'login/createuser.php' because it is ' //Validation error from empty form variables'. However, I do not have any empty fields. 

I have everything setup and in place up to that test. Curious if you can imagine what it is.

**Comment by blankbeard** [comment]
To be clear, also:

I have created the two tables (in an existing database for this project).

I have customized dbconf and config files with my relevant info.

And I have added `<?php require "login/loginheader.php"; ?>` onto each page.

I am not sure what is going wrong. This is happening when I try and test registering in my browser.

**Comment by kernelsp** [comment]
Are you letting users verify their own accounts or you sending it to an admin email to be verified?

**Comment by blankbeard** [comment]
Thank you for the reply!

I have an admin email to verify. 
It also looks like that error could be coming from the functions.php.

I'm still a little green but everything seems to look right and I didn't change anything outside of the Readme instructions. The only thing I did different is added the tables to an existing database instead of making one named "login".

Thanks for the help.

**Comment by kernelsp** [comment]
You're welcome, I'm glad to help.

Check on your 'members' table. If you already have an user registered with the same email that you've informed as admin, it won't work.

On a deeper level, it is due to a MySQL error since the column 'email' on table 'members' is an index. So basically what happens is that when you attempt to register another user with an email that already exists, you're trying to corrupt an index, thus the error.

Hope this helps.

**Comment by JanPetterMG** [comment]
About the MySQL indexes, is's supposed to be like that, there is no bugs over there.
The problem lies in the PHP code, try this instead: https://github.com/fethica/PHP-Login/issues/33#issuecomment-244650539

**Comment by kernelsp** [comment]
@JanPetterMG Hey Jan, when I said MySQL error, I didn't mean a bug on MySQL. I meant an error generated inside MySQL. Sure it is the default behavior for indexes.

The link shows a pretty good alternative.

**Comment by blankbeard** [comment]
My 'Members' sql table is completely empty. I have been trying to test it as well as add myself as a user/admin. Is there a different process I should be using to add myself first before the signup form will work? Image of what it looks like below. 

Interesting though, I decided to test it out with 
`$admin_email = '';` 
rather than my email, and now I will get the error 'Must provide a valid email address' even though there is a valid email address in the box. Image below as well.

Again, I really appreciate the help guys!!

![screen shot 2016-12-19 at 1 33 19 pm](https://cloud.githubusercontent.com/assets/16113688/21329864/c2c3c214-c5ef-11e6-9d8f-fec8a45313ad.png)
![screen shot 2016-12-19 at 1 22 56 pm](https://cloud.githubusercontent.com/assets/16113688/21329863/c2c20870-c5ef-11e6-8504-c7bc6d4ee8a4.png)

**Comment by kernelsp** [comment]
Did you apply the changes suggested on the link JanPetterMG posted (issue #33)? If yes, undo them first; I mean, get your files as they were originally.

**Comment by JanPetterMG** [comment]
Why is the email field yellow? Are you using some auto-fill (browser/extension)? Have you tried without it, or even in another browser?
Signup using that email succeed in my little test, no errors (don't worry, no emails sent).

Looks like a javascript verification issue, are you sure scripts are enabled in the browser and that the file is uploaded at the right place?
Do you find the script in the browser if you replace `signup.php` with `js/signup.js`?

Additionally, what version did you download, the `master` or `v2.0`?

**Comment by blankbeard** [comment]
No Kernelsp, I didn't make those changes cause they didn't seem like a fix for what I was experiencing. 

JanPetterMG, yeah I do have an auto fill but I tried it on Chrome, Firefox and Safari (I never use safari so it is default settings.

When I check the script in the browser I get
![screen shot 2016-12-22 at 10 38 31 pm](https://cloud.githubusercontent.com/assets/16113688/21448442/92790cd6-c897-11e6-8b83-3092da0b4734.png)

I also have a slightly changed index.php. Maybe that could be the problem?
![screen shot 2016-12-22 at 10 50 16 pm](https://cloud.githubusercontent.com/assets/16113688/21448575/33fbeeba-c899-11e6-9ce4-8661305d4e9d.png)

I am not exactly sure which one I downloaded but it was this month.

I really do appreciate this help.
